repose-aggregator/core/repose-core/src/main/java/org/openrepose/nodeservice/ReqResHeaderService.java

```diff
@@ -17,27 +17,21 @@
  * limitations under the License.
  * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
  */
-package org.openrepose.nodeservice.response;
+package org.openrepose.nodeservice;
 
-import org.openrepose.commons.utils.StringUtilities;
-import org.openrepose.core.services.headers.common.ViaHeaderBuilder;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.LoggerFactory;
 
 import javax.servlet.http.HttpServletRequest;
 
-public class ViaResponseHeaderBuilder extends ViaHeaderBuilder {
-
-    private final String reposeVersion;
-    private final String viaReceivedBy;
-
-    public ViaResponseHeaderBuilder(String reposeVersion, String configuredViaReceivedBy) {
-        this.reposeVersion = reposeVersion;
-        this.viaReceivedBy = StringUtilities.isBlank(configuredViaReceivedBy) ? "Repose" : configuredViaReceivedBy;
-    }
-
-    @Override
-    protected String getViaValue(HttpServletRequest request) {
-        StringBuilder builder = new StringBuilder(" ");
-
-        return builder.append(viaReceivedBy).append(" (Repose/").append(reposeVersion).append(")").toString();
+public interface ReqResHeaderService {
+    default void appendProtocolVersion(final StringBuilder builder, final HttpServletRequest request) {
```

Comment:
I should also note, I don't see how it's possible to get the HTTP protocol of the response from the origin service. I don't see anything in the HttpServletResponse that would provide it, and when I debugged Repose, the only mention of HTTP protocol was in the original response object's `_channel` field (`org.eclipse.jetty.server.HttpChannel`) and then in that's `_version` field.

That is, from `ResponseHeaderServiceImpl.setVia(...)`, it's in:
```
response
    .originalResponse  // private final
    .getHttpChannel()  // protected
    .getHttpVersion()  // public
```

On top of that, it's not even the correct value. It doesn't appear to actually get set to anything for responses, so it keeps its default `HTTP/1.1` value anyway.

This is why I was in WireShark earlier to verify that Repose was in fact getting an `HTTP/1.0` response. There doesn't seem to be a way for us to know that from within Repose, though, despite it being perfectly normal to send an `HTTP/1.1` request and get an `HTTP/1.0` response.